wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift

```diff
@@ -527,13 +527,16 @@ public class ZMUserSession: NSObject {
     }
 
     func createMLSClientIfNeeded() {
-        do {
-            if applicationStatusDirectory.clientRegistrationStatus.needsToRegisterMLSCLient {
-                // Make sure MLS client exists, mls public keys will be generated upon creation
-                _ = try coreCryptoProvider.coreCrypto(requireMLS: true)
+        // FIXME: [F] check async or not
+        WaitingGroupTask(context: syncContext) { [self] in
+            do {
+                if await syncContext.perform({ [self] in self.applicationStatusDirectory.clientRegistrationStatus.needsToRegisterMLSCLient }) {
+                    // Make sure MLS client exists, mls public keys will be generated upon creation
+                    _ = try await coreCryptoProvider.coreCrypto(requireMLS: true)
```

Comment:
Isn't it unnecessary to create a `Task` every time? I think we we can move the if statement outside.